### PR TITLE
CT: add CREATE2 to specification

### LINKED ISSUE #385: CT: add specification of the CREATE2 instruction
The specification of the CREATE2 instruction in Tosca CT is missing.

### PR DESCRIPTION
**Pull request by simonlechner**
This PR adds the CREATE2 opcode to the CT specification. 
Fixes #385.

### CODE REVIEW

File: go/ct/spc/specification.go [hunk 1]
```diff
@@ -1455,6 +1455,89 @@ func getAllRules() []Rule {
...
+			MemoryOffsetParameter{},
+			MemorySizeParameter{},
+			NumericParameter{},
+		},
+		effect: FailEffect().Apply,
+	})...)
+
+	rules = append(rules, rulesFor(instruction{
+		op:        CREATE2,
+		staticGas: 32000,
```
Comment (facuMH) [inline go/ct/spc/specification.go:1480]: suggestion: `name: "_non_static"`

File: go/ct/spc/specification.go [hunk 2]
```diff
@@ -1455,6 +1455,89 @@ func getAllRules() []Rule {
...
+		conditions: []Condition{
+			Eq(ReadOnly(), false),
+		},
+		parameters: []Parameter{
+			ValueParameter{},
+			MemoryOffsetParameter{},
+			MemorySizeParameter{},
+			NumericParameter{},
+		},
+		effect: func(s *st.State) {
```
Comment (facuMH) [inline go/ct/spc/specification.go:1492]: `CREATE` and `CREATE2` effects are quite similar, do you think you could unify them ?

File: go/vm/lfvm/instructions.go [hunk 3]
```diff
@@ -814,6 +814,12 @@ func opCreate2(c *context) {
 		return
 	}
 
+	// Charge for the code size
+	words := (size.Uint64() + 31) / 32
```
Comment (facuMH) [inline go/vm/lfvm/instructions.go:818]: if this change is a fix on the VM it should be a separate PR.

**Review by facuMH — CHANGES_REQUESTED** [review]

File: go/vm/lfvm/instructions.go (hunk 3 above)
Comment (simonlechner) [reply go/vm/lfvm/instructions.go:818]: There was no newly found bug in the VM implementation. But the changes performed in #446 to CREATE where also applied to CREATE2. This was done without testing the CREATE2 changes and the order had to be updated which is done in this PR.

File: go/ct/spc/specification.go [hunk 4]
```diff
@@ -1455,6 +1455,89 @@ func getAllRules() []Rule {
 		},
 	})...)
 
+	// --- CREATE2 ---
+
+	rules = append(rules, rulesFor(instruction{
+		op:        CREATE2,
+		name:      "_staticcall",
```
Comment (HerbertJordan) [inline go/ct/spc/specification.go:1462]: The term `staticcall` might be misleading since there is a `STATICCALL` instruction. How about just "_static"?

**Review by HerbertJordan — DISMISSED** [review]
Thanks, please consider the refactoring feedback provided by @facuMH before merging.

**Review by facuMH — APPROVED** [review]
thanks for the fixes !!
